src/modals/Validate.tsx

```diff
@@ -0,0 +1,227 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+import {
+    Modal,
+    Col,
+    Row,
+    TextInput,
+    Button,
+    Switch,
+    Icon
+} from 'react-materialize';
+import { Modals, SmartWalletWithBalance } from 'src/types';
+import LoadingButton from 'src/modals/LoadingButton';
+import { useStore } from 'src/context/context';
+import Utils from 'src/Utils';
+
+type ValidateProps = {
+    smartWallets: SmartWalletWithBalance[];
+    setSmartWallets: Dispatch<SetStateAction<SmartWalletWithBalance[]>>;
+    modal: Modals;
+    setModal: Dispatch<SetStateAction<Modals>>;
+};
+
+type ValidateInfo = {
+    check: boolean;
+    address: string;
+};
+
+type ValidateInfoKey = keyof ValidateInfo;
+
+function Validate(props: ValidateProps) {
+    const { smartWallets, setSmartWallets, modal, setModal } = props;
+
+    const [validate, setValidate] = useState<ValidateInfo>({
+        check: false,
+        address: ''
+    });
+
+    const [validateLoading, setValidateLoading] = useState(false);
+
+    const { state, dispatch } = useStore();
+
+    const changeValue = <T,>(value: T, prop: ValidateInfoKey) => {
+        if (!validate.check && prop === 'address' && Number(value) < 0) {
+            return;
+        }
+        setValidate((prev) => ({ ...prev, [prop]: value }));
+    };
+
+    const close = () => {
+        setModal((prev) => ({ ...prev, validate: false }));
+        setValidate({
+            check: false,
+            address: ''
+        });
+    };
+
+    const validateSmartWallets = (address: string): Boolean => {
+        for (let i = 0; i < smartWallets.length; i += 1) {
+            console.log(smartWallets[i].address, address);
+            if (smartWallets[i].address === address) {
+                alert('Smart Wallet already included');
+                return true;
+            }
+        }
+        return false;
+    };
+
+    const importSmartWallet = async () => {
+        setValidateLoading(true);
+        try {
+            dispatch({ type: 'set_loader', loader: true });
+            if (validateSmartWallets(validate.address)) {
+                dispatch({ type: 'set_loader', loader: false });
+                return;
+            }
+            const smartWallet = await state.provider!.validateSmartWallet(
+                validate.address
+            );
+            const smartWalletWithBalance = await Utils.setBalance(
+                smartWallet,
+                state.token!
+            );
+            const tempSmartWallets = [...smartWallets, smartWalletWithBalance];
+            setSmartWallets(tempSmartWallets);
+            localStorage.setItem(
+                state.account,
+                JSON.stringify(tempSmartWallets)
+            );
+            dispatch({ type: 'set_loader', loader: false });
+            close();
```

Comment:
I think that those lines should be moved after the catch, otherwise the UI keeps waiting if there are errors

Reply:
I will check it and do the necessary changes.